crates/sui-graphql-rpc/src/zklogin_verify_signature.rs

```diff
@@ -0,0 +1,199 @@
+        {
+            GenericSignature::ZkLoginAuthenticator(zk) => {
+                let bytes = FastcryptoBase64::decode(&bytes)
+                    .map_err(|_| Error::Client("Invalid base64 for bytes".to_string()))?;
+                match safe_intent {
+                    IntentScope::TransactionData => {
+                        let tx_data: TransactionData = bcs::from_bytes(&bytes)
+                            .map_err(|_| Error::Client("Invalid tx data bytes".to_string()))?;
+                        let intent_msg =
+                            IntentMessage::new(Intent::sui_transaction(), tx_data.clone());
+                        let tx_sender = tx_data.execution_parts().1;
+                        if tx_sender != author.into() {
+                            return Err(Error::Client("Tx sender mismatch author".to_string()));
+                        }
+                        match zk.verify_authenticator(
+                            &intent_msg,
+                            tx_sender,
+                            Some(curr_epoch),
+                            &verify_params,
+                        ) {
+                            Ok(_) => Ok(ZkLoginVerifyResult { errors: None }),
+                            Err(e) => Ok(ZkLoginVerifyResult {
+                                errors: Some(vec![e.to_string()]),
+                            }),
+                        }
+                    }
+                    IntentScope::PersonalMessage => {
+                        let data = PersonalMessage { message: bytes };
+                        let intent_msg = IntentMessage::new(
+                            Intent {
+                                scope: IntentScope::PersonalMessage,
+                                version: IntentVersion::V0,
+                                app_id: AppId::Sui,
+                            },
+                            data,
+                        );
+
+                        match zk.verify_authenticator(
+                            &intent_msg,
+                            author.into(),
+                            Some(curr_epoch),
+                            &verify_params,
+                        ) {
+                            Ok(_) => Ok(ZkLoginVerifyResult { errors: None }),
+                            Err(e) => Ok(ZkLoginVerifyResult {
+                                errors: Some(vec![e.to_string()]),
+                            }),
+                        }
+                    }
+                    _ => Ok(ZkLoginVerifyResult {
+                        errors: Some(vec!["Invalid intent scope".to_string()]),
+                    }),
+                }
+            }
+            _ => Ok(ZkLoginVerifyResult {
+                errors: Some(vec!["Endpoint only supports zkLogin signature".to_string()]),
+            }),
+        }
+    }
+}
+
+/// The result of the zkLogin signature verification.
+#[derive(SimpleObject, Clone)]
+pub(crate) struct ZkLoginVerifyResult {
```

Comment:
For the ease of people using this endpoint, can we add a boolean `status` field that indicates whether verification succeeded or not?

Reply:
i went back and forth on this, im in general against the pattern for wrapping a boolean in result, bc often time we assert result.is_ok() and forgot to check result.unwrap().is_true(). 

we have several ways for this
1) if its not verified, return a graphql client error, otherwise return a graphql OK result 
2) if its not verified, return a graphql OK response with boolean value and a vec of errors, otherwise return
3) if its not verified, return a graphql OK response with a vec of errors, the error is None. 

i think you are suggesting 2 and im using 3. do we prefer 1) instead?

Reply:
The caller would need to know to check that `result.data.zklogin_verify_signature.errors.length() == 0` to determine whether the verification attempt succeeded, which might not be immediately obvious - although we could spell this out in the docs. Alternatively, something like  the below is more explicit. And we're technically not just wrapping the boolean in a result since there's also a potentially empty array of string errors
```
ZkLoginVerifyResult {
  success: bool,
  errors: Vec<String>
}
```
Speaking of, it looks like the errors we write to `ZkLoginVerifyResult` are
1. Invalid intent scope
2. endpoint only supports zklogin signature
3. some `SuiError` from `zk.verify_authenticator`

Sans the third class of errors, the first two could either be considered graphql client errors, or a response state from zk login verification

Reply:
revised to use 

ZkLoginVerifyResult {
  success: bool,
  errors: Vec<String>
}